Packs/CheckpointFirewall/Integrations/CheckPointFirewallV2/CheckPointFirewallV2_test.py

```diff
@@ -103,6 +106,50 @@ def test_checkpoint_login_mechanism(mocker, session_id, integration_context, sho
         assert logout_adapter.call_count == int(should_log_in)
 
 
+def test_checkpoint_login_mechanism__invalid_restored_sid(mocker):
+    """
+    Given
+            An expired session id in context
+    When
+            Attempting to restore session ID
+    Then
+            Ensure when test_connection is called, and triggers login
+    """
+    from CheckPointFirewallV2 import main
+    port = 4434
+    mocker.patch.object(CheckPointFirewallV2.demisto, 'getIntegrationContext',
+                        return_value=INTEGRATION_CONTEXT_WITH_SID)
+    mocker.patch.object(CheckPointFirewallV2.demisto, 'command', return_value='checkpoint-host-add')
+    mocker.patch.object(CheckPointFirewallV2.Client, 'test_connection', return_value="failed")
+
+    mocker.patch.object(CheckPointFirewallV2.demisto, 'args', return_value={'session_id': None,
+                                                                            'name': 'george',
+                                                                            'ip_address': '0.0.0.0',
+                                                                            'groups': [],
+                                                                            'ignore_warnings': 'true',
+                                                                            'ignore_errors': 'false'})
+
+    mocker.patch.object(CheckPointFirewallV2.demisto, 'params', return_value={'username': {'identifier': 'user',
+                                                                                           'password': 'pass'},
+                                                                              'session_timeout': 600,
+                                                                              'domain_arg': 'domain',
+                                                                              'server': SERVER_URL,
+                                                                              'port': port})
+    mocked_address = f'{HTTPS_WWW_BASE_URL}:{port}/web_api'
+
+    import requests_mock
+    with requests_mock.Mocker() as request_mocker:
```

Comment:
you can use the `requests_mock` as a fixture, so you don't have to use the `with` statement.
Check other usages in the content repo.